core/src/test/java/com/linecorp/armeria/server/file/HttpFileTest.java

```diff
@@ -73,4 +80,41 @@ void redirect() throws Exception {
         assertThat(agg.status()).isEqualTo(HttpStatus.TEMPORARY_REDIRECT);
         assertThat(agg.headers().get(HttpHeaderNames.LOCATION)).isEqualTo("/foo/bar?a=b");
     }
+
+    @Test
+    void createFromFileUrl(@TempDir Path tempDir) throws Exception {
+        final URL file = tempDir.resolve("test.txt").toUri().toURL();
+        final HttpFileBuilder builder = HttpFile.builder(file);
+        assertThat(builder).isInstanceOf(HttpFileBuilder.FileSystemHttpFileBuilder.class);
+    }
+
+    @Test
+    void createFromHttpUrl() throws Exception {
+        final URL url = new URL("https://line.me");
+        assertThatThrownBy(() -> HttpFile.builder(url)).isInstanceOf(IllegalArgumentException.class).hasMessageContaining("Unsupported URL: https://line.me (must start with 'file:', 'jar:file' or 'jrt:')");
+    }
+
+    @Test
+    // in JDK 8, getting class by getResource returns jar protocol
+    @DisabledOnJre(JRE.JAVA_8)
+    void createFromJrtUrl() {
+        final URL jarFileUrl = ClassLoader.getSystemClassLoader().getResource("java/lang/Object.class");
+        assertThat(jarFileUrl.getProtocol()).isEqualTo("jrt");
+        final HttpFileBuilder builder = HttpFile.builder(jarFileUrl);
+        assertThat(builder).isInstanceOf(HttpFileBuilder.ClassPathHttpFileBuilder.class);
+    }
+
+    @Test
+    void createFromJarFileUrl() {
+        final URL jarFileUrl = Test.class.getClassLoader().getResource("META-INF/LICENSE.md");
+        assertThat(jarFileUrl.getProtocol()).isEqualTo("jar");
+        final HttpFileBuilder builder = HttpFile.builder(jarFileUrl);
+        assertThat(builder).isInstanceOf(HttpFileBuilder.ClassPathHttpFileBuilder.class);
+    }
+
+    @Test
+    void createFromJarHttpUrl() throws Exception {
+        final URL jarHttpUrl = new URL("jar:http://www.foo.com/bar/baz.jar!/COM/foo/Quux.class");
+        assertThatThrownBy(() -> HttpFile.builder(jarHttpUrl)).isInstanceOf(IllegalArgumentException.class).hasMessageContaining("Unsupported URL: jar:http://www.foo.com/bar/baz.jar!/COM/foo/Quux.class (must start with 'file:', 'jar:file' or 'jrt:')");
```

Comment:
Ditto. This exceeds the line length limit.
https://github.com/line/armeria/blob/94a1bf40ff41e1cfa9cd0c1a08f7bde14a7153c9/settings/checkstyle/checkstyle.xml#L27